src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,93 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
+import { Movie } from '../../types/Movie';
 import { TextField } from '../TextField';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void,
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [newTitle, setNewTitle] = useState('');
+  const [newDescription, setNewDescription] = useState('');
+  const [newImgUrl, setNewImgUrl] = useState('');
+  const [newImdbUrl, setNewImdbUrl] = useState('');
+  const [newImdbId, setNewImdbId] = useState('');
+  const requiredFields = (
+    newTitle
+    && newImgUrl
+    && newImdbUrl
+    && newImdbId
+  );
+
+  useEffect(() => {
+    setNewTitle('');
+    setNewDescription('');
+    setNewImgUrl('');
+    setNewImdbUrl('');
+    setNewImdbId('');
+  }, [count]);
+
+  const newMovie: Movie = {
+    title: newTitle,
+    description: newDescription,
+    imgUrl: newImgUrl,
+    imdbUrl: newImdbUrl,
+    imdbId: newImdbId,
+  };
+
+  const onSubmitHandler = () => {
+    onAdd(newMovie);
+    setCount((state) => state + 1);
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={onSubmitHandler}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={newTitle}
+        onChange={setNewTitle}
         required
       />
 
       <TextField
         name="description"
         label="Description"
-        value=""
+        value={newDescription}
+        onChange={setNewDescription}
       />
 
       <TextField
         name="imgUrl"
         label="Image URL"
-        value=""
+        value={newImgUrl}
+        onChange={setNewImgUrl}
```

Comment:
```suggestion
        onChange={(newValue) => setNewImgUrl(newValue)}
```
I know that it works the other way also but this makes it more understandable.